src/codegen/codegen_c_visitor.cpp

```diff
@@ -2484,100 +2500,161 @@ void CodegenCVisitor::print_mechanism_global_var_structure() {
             auto name = var->get_name();
             auto length = var->get_length();
             if (var->is_array()) {
-                printer->add_line(fmt::format("{}{} {}[{}];", qualifier, float_type, name, length));
+                printer->fmt_line("{}{} {}[{}] /* TODO init top-local-array */;",
+                                  qualifier,
+                                  float_type,
+                                  name,
+                                  length);
             } else {
-                printer->add_line(fmt::format("{}{} {};", qualifier, float_type, name));
+                printer->fmt_line("{}{} {} /* TODO init top-local */;",
+                                  qualifier,
+                                  float_type,
+                                  name);
             }
             codegen_global_variables.push_back(var);
         }
     }
 
     if (!info.thread_variables.empty()) {
-        printer->add_line(fmt::format("{}int thread_data_in_use;", qualifier));
-        printer->add_line(
-            fmt::format("{}{} thread_data[{}];", qualifier, float_type, info.thread_var_data_size));
+        printer->fmt_line("{}int thread_data_in_use{};", qualifier, value_initialise);
+        printer->fmt_line("{}{} thread_data[{}] /* TODO init thread_data */;",
+                          qualifier,
+                          float_type,
+                          info.thread_var_data_size);
         codegen_global_variables.push_back(make_symbol("thread_data_in_use"));
         auto symbol = make_symbol("thread_data");
         symbol->set_as_array(info.thread_var_data_size);
         codegen_global_variables.push_back(symbol);
     }
 
-    printer->add_line(fmt::format("{}int reset;", qualifier));
+    // TODO: remove this entirely?
```

Comment:
Bit of background:

"thread" variables are a way in neuron to avoid race condition when variables are being updated by multiple threads (e.g. user declare something `GLOBAL` and nrnivmodl converts it to "thread variable" with `#threads` copies)

In context of coreneuron, this is not sufficient due to SIMD/SIMT execution. One solution I am thinking is that NEURON should convert those variables to "RANGE" in order to avoid race conditions.

Reply:
I think this comment was referring to `reset`, which never seems to be touched in code generated by NMODL.